src/bundle/Controller/LocationController.php

```diff
@@ -144,12 +144,16 @@ private function deleteLocationsBasedOnFormSubmit(FormInterface $form, $redirect
         return false;
     }
 
-    private function addLocationBasedOnFormSubmit(FormInterface $form, Content $content)
+    private function addLocationsBasedOnFormSubmit(FormInterface $form, Content $content)
     {
         if ($form->get('add')->isClicked()) {
-            $this->uiLocationService->addLocation(
-                $content->contentInfo,
-                $form->get('parentLocationId')->getData()
+            $locationService = $this->uiLocationService;
+            $contentInfo = $content->contentInfo;
+            array_map(
+                function ($locationId) use ($locationService, $contentInfo) {
+                    $locationService->addLocation($contentInfo, (int) $locationId);
+                },
+                explode(',', $form->get('parentLocationId')->getData())
```

Comment:
I'm not involved directly into hybryd ui development, but I think that this kind of logic should be moved from controller to a Form Definition level as DataTransformer. Skip this comment if there are other assumptions in project.